src/main/java/io/bosch/bookLibrary/service/impl/UserServiceImpl.java

```
@@ -0,0 +1,76 @@
+package io.bosch.bookLibrary.service.impl;
+
+import io.bosch.bookLibrary.entity.UserEntity;
+import io.bosch.bookLibrary.entity.dto.UserDTO;
+import io.bosch.bookLibrary.repository.UserRepository;
+import io.bosch.bookLibrary.service.UserService;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    private final UserRepository userRepository;
+    private final ModelMapper modelMapper;
+
+    public UserServiceImpl(UserRepository userRepository, ModelMapper modelMapper) {
+        this.userRepository = userRepository;
+        this.modelMapper = modelMapper;
+    }
+
+    @Override
+    public List<UserDTO> getAllUsers() {
+        return userRepository.findAll()
+                .stream()
+                .map(this::asUser)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public void deleteBook(long id) {
+        Optional<UserEntity> userEntity = userRepository.findById(id);
+        if (userEntity.isPresent()) {
+            userRepository.delete(userEntity.get());
+        } else {
+            //TODO
```

You can throw an exception if an entity does not exists